### PR TITLE
fix(config): respect `api_server.workers`

### PR DESCRIPTION
**Pull request by judahrand**
## What does this PR address?

`bentoml serve . --production` was ignoring `api_server.workers` in `bento_configuration.yaml`. This is now fixed. I believe I've maintained the previous behaviour when the value was not manually set by the user.

Fixes #(issue)

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, both `make format` and `make lint` script have passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: bentoml/_internal/configuration/default_configuration.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 api_server:
-  workers: 1
+  workers: null  # When this is set to null the number of available CPU cores is used.
```
Comment (sauyon) [inline bentoml/_internal/configuration/default_configuration.yaml:2]:
```suggestion
  workers: ~ # When this is set to null the number of available CPU cores is used.
```

**Review by sauyon — COMMENTED** [review]
One very minor nit, but otherwise looks great, thank you!

**Review by aarnphm — APPROVED** [review]
Thanks for fixing this

**Review by ssheng — APPROVED** [review]
Great catch and very lean fix. Thanks, @judahrand.

**Comment by aarnphm** [comment]
I will make a follow up PR to address some pylint issue.

**Pull request merged**